buttercup.el

```diff
@@ -929,9 +937,110 @@ Calls either `buttercup-reporter-batch' or
+             (append buttercup-reporter-batch--failures
+                     (list arg))))
+      ((eq (buttercup-spec-status arg) 'pending)
+       (if (equal (buttercup-spec-failure-description arg) "SKIPPED")
+           (buttercup--print "  %s\n" (buttercup-spec-failure-description arg))
+         (buttercup--print (buttercup-colorize "\r%s%s  %s\n" 'yellow)
+                           (make-string (* 2 level) ?\s)
+                           (buttercup-spec-description arg)
+                           (buttercup-spec-failure-description arg))))
+      (_
+       (error "Unknown spec status %s" (buttercup-spec-status arg))))))
+
+    (`buttercup-done
+     (dolist (failed buttercup-reporter-batch--failures)
+       (let ((description (buttercup-spec-failure-description failed))
+             (stack (buttercup-spec-failure-stack failed)))
+         (buttercup--print "%s\n" (make-string 40 ?=))
+         (buttercup--print (buttercup-colorize "%s\n" 'red) (buttercup-spec-full-name failed))
+         (when stack
+           (buttercup--print "\nTraceback (most recent call last):\n")
+           (dolist (frame stack)
+             (let ((line (format "  %S" (cdr frame))))
+               (when (> (length line) 79)
+                 (setq line (concat (substring line 0 76)
+                                    "...")))
+               (buttercup--print "%s\n" line))))
+         (cond
+          ((stringp description)
+           (buttercup--print (concat (buttercup-colorize "FAILED" 'red ) ": %s\n")
+                             description))
+          ((eq (car description) 'error)
+           (buttercup--print "%S: %S\n\n"
+                             (car description)
+                             (cadr description)))
+          (t
+           (buttercup--print "FAILED: %S\n" description)))
+         (buttercup--print "\n")))
+     (let ((defined (buttercup-suites-total-specs-defined arg))
+           (pending (buttercup-suites-total-specs-pending arg))
+           (failed (buttercup-suites-total-specs-failed arg))
+           (duration (- (float-time)
+                        buttercup-reporter-batch--start-time)))
+       (if (> pending 0)
+           (buttercup--print
+            (concat
+             "Ran %s out of %s specs,"
+             (buttercup-colorize " %s failed" (if (eq 0 failed) 'green 'red))
+             ", in %.1f seconds.\n")
+            (- defined pending)
+            defined
+            failed
+            duration)
+         (buttercup--print
+          (concat
+           "Ran %s specs,"
+           (buttercup-colorize " %s failed" (if (eq 0 failed) 'green 'red))
+           ", in %.1f seconds.\n"
+           )
+          defined
+          failed
+          duration))
+       (when (> failed 0)
+         (error ""))))
+
+    (_
+     ;; Fall through to buttercup-reporter-batch implementation.
+     (buttercup-reporter-batch event arg)))
+  )
+
 (defun buttercup--print (fmt &rest args)
```

Comment:
Well, depends on what you think. That was my initial idea, until I realized that in color mode it needs to re-print the spec name, with color, over the colorless version printed when the spec started. So it prints a carriage return at the beginning of the new string.

So it's not just a matter of adding colors, the strings are quite different in the color/no-color cases.

That's why I started out creating a new reporter, but it could be merged into the standard reporter, thought it would beef it up a bit.